### PR TITLE
trivial changes

### PR DESCRIPTION
**Pull request by chenbd**
"h2o_balancer_t" type defined twice
"__builtin_expect" under clang
remove "inline"

### CODE REVIEW

File: include/h2o/memory.h [hunk 1]
```diff
@@ -38,7 +38,7 @@ extern "C" {
 #define H2O_STRUCT_FROM_MEMBER(s, m, p) ((s *)((char *)(p)-offsetof(s, m)))
 #define H2O_ALIGNOF(type) (__alignof__(type))
 
-#if __GNUC__ >= 3
+#if (__GNUC__ >= 3) || defined(__clang__)
```
Comment (kazuho) [inline include/h2o/memory.h]: Cousin you please revert the change? We can and do rely at other parts of the source code on clang setting __GNUC__.

Other changes look fine to me.

Comment (chenbd) [reply include/h2o/memory.h]: yes, reverted.

**Comment by kazuho** [comment]
Thank you for the changes!
